crypto/asn1/a_mbstr.c

```diff
@@ -349,47 +349,13 @@ static int is_printable(unsigned long value)
     if (value > 0x7f)
         return 0;
     ch = (int)value;
-    /*
-     * Note: we can't use 'isalnum' because certain accented characters may
-     * count as alphanumeric in some environments.
-     */
-#ifndef CHARSET_EBCDIC
-    if ((ch >= 'a') && (ch <= 'z'))
+    if (ossl_isalnum(ch) || strchr("'()+,-./:=?", ch))
         return 1;
-    if ((ch >= 'A') && (ch <= 'Z'))
-        return 1;
-    if ((ch >= '0') && (ch <= '9'))
-        return 1;
-    if ((ch == ' ') || strchr("'()+,-./:=?", ch))
-        return 1;
-#else                           /* CHARSET_EBCDIC */
-    if ((ch >= os_toascii['a']) && (ch <= os_toascii['z']))
-        return 1;
-    if ((ch >= os_toascii['A']) && (ch <= os_toascii['Z']))
-        return 1;
-    if ((ch >= os_toascii['0']) && (ch <= os_toascii['9']))
-        return 1;
-    if ((ch == os_toascii[' ']) || strchr("'()+,-./:=?", os_toebcdic[ch]))
-        return 1;
-#endif                          /* CHARSET_EBCDIC */
     return 0;
 }
 
 /* Return 1 if the character is a digit or space */
 static int is_numeric(unsigned long value)
 {
-    int ch;
-    if (value > 0x7f)
-        return 0;
-    ch = (int)value;
-#ifndef CHARSET_EBCDIC
-    if (!isdigit(ch) && ch != ' ')
-        return 0;
-#else
-    if (ch > os_toascii['9'])
-        return 0;
-    if (ch < os_toascii['0'] && ch != os_toascii[' '])
-        return 0;
-#endif
-    return 1;
+    return ossl_isdigit(value) || value == ' ';
```

Comment:
This is an ASN1-type string, which uses ASCII/UTF/etc encoding, space has a different value in EBCDIC than ASCI/UTF8. When this is is compiled on an EBCDIC machine, the value comparison will be to the wrong space, which is why the original code used os_toascii[].